### DISCUSSION TITLE
Preventing an onClick after a drag

### DISCUSSION THREAD

**Discussion by rahulbrahmal** [xyflow/xyflow#1412] [Q&A]
Hi!

In my current project, when I drag a node, at the end of the drag (when I "unclick" the mouse), an onClick action is fired. This is causing some unintended behaviour in my board, as I am performing an action to the block clicked, and this is firing every time the block is dragged.

Is there a way to prevent the onClick event from firing during a drag?

**Comment by Azim-Ahmed** [comment]
you can use onDrop function. Let check if it is working or not.


`  const onDrop = (event) => {
    event.preventDefault();
    const reactFlowBounds = reactFlowWrapper.current.getBoundingClientRect();
    const type = event.dataTransfer.getData('application/reactflow');
    const position = reactFlowInstance.project({
      x: event.clientX - reactFlowBounds.left,
      y: event.clientY - reactFlowBounds.top,
    });

    //new node Data
    const newNode = {
      type,
      position,
      data: {
        label: `${type} node`,
      },
    };
    setElements((es) => es.concat(newNode));
  };
